std is included even in no_std environments.

I am trying to write a piece of code targeting the x86_64 ABI using a custom target, which means there's no `std` available for my build target so I am building with `#[no_std]`. It seems that even with the std flag disabled, the package still tries to pull an `std` package into the build. The following is the build error:

```bash
$ xargo build --release --target=raw
   Compiling pkg-config v0.3.9
   Compiling libc v0.2.36
   Compiling rlibc v1.0.0
   Compiling compiler_builtins v0.1.0 (https://github.com/rust-lang-nursery/compiler-builtins#0ba07e49)
error[E0463]: can't find crate for `std`
  |
  = note: the `raw` target may not be installed

error: aborting due to previous error

```

Note I am not actually doing anything with the package yet, if I remove it from my dependencies the build succeeds and no `std` is ever built.

My environment:

#### Cargo.toml
```toml
# ...
[dependencies.sodiumoxide]                                                                                                                                                                                                                                                    
version = "*"                                                                                                                                                                                                                                                                 
default-features = false     
features=[]
```

#### raw.json
```json
  {                                                                                                                                                                                                                                                                             
    "pre-link-args"                    : ["-m64", "-nostdlib", "-static"],                                                                                                                                                                                                      
    "dynamic-linking"                  : false,                                                                                                                                                                                                                                 
    "relocation-model"                 : "pic",                                                                                                                                                                                                                                 
    "no-compiler-rt"                   : true,                                                                                                                                                                                                                                  
    "no-default-libraries"             : true,                                                                                                                                                                                                                                  
    "position-independent-executables" : false,                                                                                                                                                                                                                                 
    "llvm-target"                      : "x86_64-unknown-none",                                                                                                                                                                                                                 
    "data-layout"                      : "e-m:e-i64:64-f80:128-n8:16:32:64-S128",                                                                                                                                                                                               
    "linker-flavor"                    : "gcc",                                                                                                                                                                                                                                 
    "target-endian"                    : "little",                                                                                                                                                                                                                              
    "target-pointer-width"             : "64",                                                                                                                                                                                                                                  
    "target-c-int-width"               : "32",                                                                                                                                                                                                                                  
    "arch"                             : "x86_64",                                                                                                                                                                                                                              
    "os"                               : "none",                                                                                                                                                                                                                                
    "disable-redzone"                  : true,                                                                                                                                                                                                                                  
    "features"                         : "+avx,+mmx,+sse",                                                                                                                                                                                                                      
    "panic-strategy"                   : "abort"                                                                                                                                                                                                                                
  }
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,18 +16,18 @@ exclude = [
 ]
 
 [badges]
-travis-ci = { repository = "dnaq/sodiumoxide" }
+travis-ci = { repository = "sodiumoxide/sodiumoxide" }
 
 [dependencies]
-libc = "0.2"
+libc = { version = "^0.2.41" , default-features = false }
 libsodium-sys = { version = "0.0.16", path = "libsodium-sys" }
-serde = { version="1", optional = true }
+serde = { version = "^1.0.59", default-features = false, optional = true }
 
 [dev-dependencies]
-serde = "1"
-serde_json = "1"
-rustc-serialize = "0.3"
-rmp-serde = "0.13"
+serde = "^1.0.59"
+serde_json = "^1.0.17"
+rustc-serialize = "^0.3.24"
+rmp-serde = "^0.13.7"
 
 [features]
 benchmarks = []

--- a/libsodium-sys/Cargo.toml
+++ b/libsodium-sys/Cargo.toml
@@ -12,13 +12,13 @@ categories = ["cryptography", "api-bindings"]
 version = "0.0.16"
 
 [badges]
-travis-ci = { repository = "dnaq/sodiumoxide" }
+travis-ci = { repository = "sodiumoxide/sodiumoxide" }
 
 [build-dependencies]
-pkg-config = "0.3"
+pkg-config = "^0.3.11"
 
 [dependencies]
-libc = "0.2"
+libc = { version = "^0.2.41" , default-features = false }
 
 [lib]
 name = "libsodium_sys"

--- a/libsodium-sys/lib.rs
+++ b/libsodium-sys/lib.rs
@@ -1,4 +1,5 @@
 #![allow(non_upper_case_globals)]
+#![no_std]
 
 extern crate libc;
 use libc::{c_void, c_int, c_ulonglong, c_char, size_t, uint64_t};

